backend/app/services/memory/client.py

```
@@ -0,0 +1,382 @@
+            base_url: mem0 service base URL (default: from settings)
+            api_key: Optional API key (default: from settings)
+            timeout: Default HTTP timeout in seconds (default: from settings)
+
+        Raises:
+            ValueError: If base_url is empty or None
+        """
+        raw_base = base_url or settings.MEMORY_BASE_URL
+        if not raw_base:
+            raise ValueError(
+                "base_url cannot be empty. Provide a valid mem0 service URL."
+            )
```

Ruff (0.14.11)

Avoid specifying long messages outside the exception class

(TRY003)